shakenfist/daemons/queues.py

```diff
@@ -29,55 +35,68 @@ def handle(jobname, workitem):
     task = None
     try:
         for task in workitem.get('tasks', []):
-            instance_uuid = task.get('instance_uuid')
-            log_i = log.withInstance(instance_uuid)
-            log_i.info("Starting task")
+            if InstanceTask.__subclasscheck__(type(task)):
+                instance_uuid = task.instance_uuid()
+                log_i = log.withInstance(instance_uuid)
```

Comment:
I think I'd rather just replace log here instead of creating log_i.

Reply:
`log` cannot be replaced because there is no method to remove a log field. A task may not have an `instance_uuid`.

Reply:
But everyone below uses log_i right? So wouldn't we just be replacing the value of log in this scope?

Reply:
The `withInstance()` adds the `instance_uuid` field. So if - as you suggest - we store that logger with the added instance field in `log`. Next loop we call `withInstance()` again with the new instance_uuid and it will overwrite the field in the logger. All good. What happens when we loop a third time and do not have an instance_uuid (because the task doesn't have an instance_uuid). Overwrite the field with `None`? Then we have a incorrect log output implying that the field was supplied with None.

I'm wondering if you are assuming that instance_uuid is always the same in `workitem`? Not an incorrect assumption but if true we should maybe change the data structure?

Am I being totally obtuse?

[Sneak peak] In either case, it will be moot because I will be suggesting in a near-future PR that `workitem`s only contain single tasks and cause single state transitions.

Reply:
I think this is me being asleep at the wheel and forgetting this code is in a loop. I blame the github PR viewer thing not showing enough context.